internal/core/api.rs

```diff
@@ -40,11 +40,11 @@ impl LogicalPosition {
 
     /// Convert this logical position to a physical position by multiplying the coordinates with the
     /// specified scale factor.
-    pub fn to_physical(&self, scale_factor: f32) -> PhysicalPosition {
-        PhysicalPosition::from_logical(*self, scale_factor)
+    pub fn to_physical(self, scale_factor: f32) -> PhysicalPosition {
```

Comment:
This, and the other in this file, are part of the public API.   I wonder if it's safe to do this change.

Reply:
Right. I think it's OK.

To be able to call `to_physical()`, users had to have reference on types that are and were copy.
So all call to `to_physical()` will still work as they will now copy if needed.

The following was working and still is.

```rust
let a = LogicalPosition { x: 1.0, y: 1.0 };
let x = a.to_physical(); // was taking &self, is now taking self takes to Copy
```

The only thing I see is that some `needless_borrow` warning could be added, but the chances are pretty low I think.

Reply:
Unless someone did:

```rust
let a = LogicalPosition { x: 1.0, y: 1.0 };
LogicalPosition::to_physical(&a, 1.0); 
```

But perhaps nobody does that?

Reply:
Rarely I think, and that is the case where they would get the clippy warning. It would still work.

Reply:
Actually it does not work 😢

Reply:
I believe it's better to #[allow] them for now.
We can open an issue about API changes we can do in Slint 2

Reply:
Actually, when fixing this warning I when further that just what clippy was warning about.

To fix the warning I only need to change two `to_euclid()` which are `pub(crate)`  so this is not breaking.
I removed the other changes I made to align things that was not need and was breaking.

Reply:
Filled https://github.com/slint-ui/slint/issues/3023